drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c

```diff
@@ -578,10 +578,17 @@ static void job_abort(void *priv)
 	ctx->aborting = 1;
 }
 
-static inline unsigned int get_sizeimage(int bpl, int height,
+static inline unsigned int get_sizeimage(int bpl, int width, int height,
 					 struct bcm2835_codec_fmt *fmt)
 {
-	return (bpl * height * fmt->size_multiplier_x2) >> 1;
+	if (fmt->flags & V4L2_FMT_FLAG_COMPRESSED) {
+		if (width * height > 1280 * 720)
+			return DEF_COMP_BUF_SIZE_GREATER_720P;
+		else
+			return DEF_COMP_BUF_SIZE_720P_OR_LESS;
```

Comment:
I pressume there is no need to consider the bits per pixel here?

Reply:
Sorry, bytes per line.

Reply:
bytesperline will be 0 for any compressed format. That's the reason I've had to add in the width parameter to the function.

Spec - https://www.linuxtv.org/downloads/v4l-dvb-apis-new/uapi/v4l/pixfmt-v4l2.html
> For compressed formats the bytesperline value makes no sense. Applications and drivers must set this to 0 in that case.